### PR TITLE
Populate sources and sorts for basic search queries on refresh

### PR DESCRIPTION
**Pull request by samuelechu**
#302

### CODE REVIEW

File: src/main/webapp/components/basic-search/basic-search.js [hunk 1]
```diff
@@ -335,6 +351,14 @@ export const BasicSearchQueryBuilder = props => {
           )
         })
         .valueSeq()}
+      <FilterCard label="Sources">
```
Comment (samuelechu) [inline src/main/webapp/components/basic-search/basic-search.js:354]: This makes sources a mandatory filter, like it is in advanced search.

File: src/main/webapp/components/basic-search/basic-search.js [hunk 2]
```diff
@@ -233,26 +233,42 @@ const defaultFilters = {
...
   let sorts = get(filterMap, 'sortOrder')
-  if (sorts) {
-    sorts = sorts.map(
-      ({ propertyName, sortOrder }) => `${propertyName},${sortOrder}`
-    )
-  }
+  sorts = sorts
+    ? sorts.map(({ propertyName, sortOrder }) => `${propertyName},${sortOrder}`)
+    : null
+  let sources = get(filterMap, 'sources')
```
Comment (willwill96) [inline src/main/webapp/components/basic-search/basic-search.js:267]: :pencil2: const

File: src/main/webapp/components/basic-search/basic-search.js [hunk 3]
```diff
@@ -233,26 +233,42 @@ const defaultFilters = {
...
+
   return {
     filterTree: toFilterTree(filterMap),
-    sources: get(filterMap, 'sources'),
+    sources: sources ? sources : null,
```
Comment (willwill96) [inline src/main/webapp/components/basic-search/basic-search.js:271]: :pencil2: could also be written 
```suggestion
    sources: sources || null,
```

File: src/main/webapp/components/basic-search/basic-search.js [hunk 4]
```diff
@@ -213,15 +213,15 @@ const filters = {
...
+  //sources: BasicSources,
   sortOrder: BasicSortOrder,
 }
 
 const filterLabels = {
   [LOCATION_KEY]: 'Location',
   timeRange: 'Time Range',
   [MATCHTYPE_KEY]: 'Match Types',
-  sources: 'Sources',
+  //sources: 'Sources',
```
Comment (willwill96) [inline src/main/webapp/components/basic-search/basic-search.js:224]: delete comment? also on line 216

File: src/main/webapp/components/basic-search/basic-search.js [hunk 5]
```diff
@@ -233,26 +233,42 @@ const defaultFilters = {
...
-    return fromFilterTree(props.query.filterTree, {
-      basicSearchMatchType: props.basicSearchMatchType,
+const getFilterMap = ({ query, basicSearchMatchType }) => {
+  if (!query) {
+    return Map({ text: '*', sources: undefined })
+  }
+
+  let filterMap = query.filterTree
+    ? fromFilterTree(query.filterTree, {
+        basicSearchMatchType: basicSearchMatchType,
```
Comment (willwill96) [inline src/main/webapp/components/basic-search/basic-search.js:243]: :pencil2: could use shorthand here 
```suggestion
        basicSearchMatchType,
```

File: src/main/webapp/components/basic-search/basic-search.js [hunk 6]
```diff
@@ -233,26 +233,42 @@ const defaultFilters = {
   [LOCATION_KEY]: makeDefaultSearchGeo(),
 }
 
-const getFilterMap = props => {
-  if (props.query && props.query.filterTree) {
-    return fromFilterTree(props.query.filterTree, {
-      basicSearchMatchType: props.basicSearchMatchType,
+const getFilterMap = ({ query, basicSearchMatchType }) => {
+  if (!query) {
+    return Map({ text: '*', sources: undefined })
```
Comment (willwill96) [inline src/main/webapp/components/basic-search/basic-search.js:238]: :pencil2: seems equivalent to
```suggestion
    return Map({ text: '*' })
```

**Pull request merged**